diffrax/_brownian/tree.py

```diff
@@ -369,14 +429,114 @@ def _body_fun(_state: _State):
+            # Single-seed generation of Brownian paths and integrals
+            assert final_state.bhh_s_u_su is not None
+            assert final_state.bkk_s_u_su is not None
+            bhh_s, bhh_u, bhh_su = final_state.bhh_s_u_su
+            bkk_s, bkk_u, bkk_su = final_state.bkk_s_u_su
+
+            su3 = jnp.power(su, 3)
+            sr_by_su = sr / su
+            sr_by_su_3 = jnp.power(sr_by_su, 3)
+            sr_by_su_5 = jnp.power(sr_by_su, 5)
+            ru_by_su = ru / su
+            sr_ru_by_su2 = sr_by_su * ru_by_su
+            sr2 = jnp.square(sr)
+            ru2 = jnp.square(ru)
+            su2 = jnp.square(su)
+
+            # compute the mean of (W_sr, H_sr, K_sr) conditioned on
+            # (W_s, H_s, K_s, W_u, H_u, K_u)
+            with jax.numpy_dtype_promotion("standard"):
+                bb_mean = (6 * sr_ru_by_su2 / su) * bhh_su + (
+                    120 * sr_ru_by_su2 * (0.5 - sr_by_su) / su2
+                ) * bkk_su
+                w_mean = sr_by_su * w_su + bb_mean
+                h_mean = (sr_by_su**2 / su) * (bhh_su + (30 * ru_by_su / su) * bkk_su)
+                k_mean = (sr_by_su_3 / su2) * bkk_su
+
+                # compute the covariance matrix of (W_sr, H_sr, K_sr) conditioned on
+                # (W_s, H_s, K_s, W_u, H_u, K_u)
+                ww_cov = (
+                    sr_by_su * ru_by_su * ((sr - ru) ** 4 + 4 * (sr2 * ru2))
+                ) / su3
+                wh_cov = -(sr_by_su_3 * ru_by_su * (sr2 - 3 * sr * ru + 6 * ru2)) / (
+                    2 * su
+                )
+                wk_cov = (sr_by_su**4) * ru_by_su * (sr - ru) / 12
+                hh_cov = (
+                    sr / 12 * (1 - sr_by_su_3 * (sr2 + 2 * sr * ru + 16 * ru2) / su2)
+                )
+                hk_cov = -(ru / 24) * sr_by_su_5
+                kk_cov = (sr / 720) * (1.0 - sr_by_su_5)
+
+            cov = jnp.array(
+                [
+                    [ww_cov, wh_cov, wk_cov],
+                    [wh_cov, hh_cov, hk_cov],
+                    [wk_cov, hk_cov, kk_cov],
+                ]
+            )
+
+            if self._spline == "sqrt":
+                # NOTE: not compatible with jnp.float16
+                hat_y = jr.multivariate_normal(
+                    final_state.key,
+                    jnp.zeros((3,), dtype),
+                    cov,
+                    shape=shape,
+                    dtype=dtype,
+                    method="svd",
+                )
+
+            elif self._spline == "zero":
+                hat_y = jnp.zeros(shape=shape + (3,), dtype=dtype)
+            else:
+                raise ValueError(
+                    f"When levy_area='space-time-time', only 'sqrt' and"
+                    f" 'zero' splines are permitted, got {self._spline}."
+                )
+
+            hat_w_sr, hat_hh_sr, hat_kk_sr = [
+                x.squeeze(axis=-1) for x in jnp.split(hat_y, 3, axis=-1)
+            ]
```

Comment:
I thnk just `hat_w_sr, hat_hh_sr, hat_kk_sr = hat_y` will work?

Reply:
No, I don't think it will, because we're splitting by the last dimension, whereas tuple unpacking splits along the first.